lib/modules/manager/dockerfile/extract.spec.ts

```diff
@@ -647,6 +649,343 @@ describe('modules/manager/dockerfile/extract', () => {
         ]
       `);
     });
+
+    it('handles implausible line continuation', () => {
+      const res = extractPackageFile(
+        'FROM alpine:3.5\n\nRUN something \\'
+      ).deps;
+      expect(res).toMatchInlineSnapshot(`
+        Array [
+          Object {
+            "autoReplaceStringTemplate": "{{depName}}{{#if newValue}}:{{newValue}}{{/if}}{{#if newDigest}}@{{newDigest}}{{/if}}",
+            "currentDigest": undefined,
+            "currentValue": "3.5",
+            "datasource": "docker",
+            "depName": "alpine",
+            "depType": "final",
+            "replaceString": "alpine:3.5",
+          },
+        ]
+      `);
+    });
+
+    it('handles multi-line FROM with space after escape character', () => {
+      const res = extractPackageFile('FROM \\ \nnginx:1.20\n').deps;
+      expect(res).toMatchInlineSnapshot(`
+        Array [
+          Object {
+            "autoReplaceStringTemplate": "{{depName}}{{#if newValue}}:{{newValue}}{{/if}}{{#if newDigest}}@{{newDigest}}{{/if}}",
+            "currentDigest": undefined,
+            "currentValue": "1.20",
+            "datasource": "docker",
+            "depName": "nginx",
+            "depType": "final",
+            "replaceString": "nginx:1.20",
+          },
+        ]
+      `);
+    });
+
+    it('handles FROM without ARG default value', () => {
+      const res = extractPackageFile('ARG img_base\nFROM $img_base\n').deps;
+      expect(res).toMatchInlineSnapshot(`
+        Array [
+          Object {
+            "autoReplaceStringTemplate": "{{depName}}{{#if newValue}}:{{newValue}}{{/if}}{{#if newDigest}}@{{newDigest}}{{/if}}",
+            "datasource": "docker",
+            "depType": "final",
+            "replaceString": "$img_base",
+            "skipReason": "contains-variable",
+          },
+        ]
+      `);
+    });
+
+    it('handles FROM with empty ARG default value', () => {
+      const res = extractPackageFile(
+        'ARG patch1=""\nARG patch2=\nFROM nginx:1.20${patch1}$patch2\n'
+      ).deps;
+      expect(res).toMatchInlineSnapshot(`
+        Array [
+          Object {
+            "currentDigest": undefined,
+            "currentValue": "1.20",
+            "datasource": "docker",
+            "depName": "nginx",
+            "depType": "final",
+            "replaceString": "FROM nginx:1.20\${patch1}$patch2",
+          },
+        ]
+      `);
+    });
+
+    it('handles FROM with version in ARG value', () => {
+      const res = extractPackageFile(
+        'ARG\tVARIANT="1.60.0-bullseye"\nFROM\trust:${VARIANT}\n'
+      ).deps;
+      expect(res).toMatchInlineSnapshot(`
+        Array [
+          Object {
+            "currentDigest": undefined,
+            "currentValue": "1.60.0-bullseye",
+            "datasource": "docker",
+            "depName": "rust",
+            "depType": "final",
+            "replaceString": "ARG\tVARIANT=\\"1.60.0-bullseye\\"",
```

Comment:
shouldn't we add an auto replace template to support digest pinning?

Reply:
You're right. I just added another commit that supplements `autoReplaceStringTemplate`. 
Test repository: https://github.com/Churro/renovate-dockerfile-test3/pulls